File: SECOM_AJIS/Scripts/Contract/CTS054.js
/*--- Main ---*/
$(document).ready(function () {
    EnableRegisterCommand();
    /*==== Attach Datetime Picker ====*/
    $("#dpExpectOperationDate").InitialDate();
    $("#ContractChange").SetViewMode(false);

    $("#chkIsimportantCustomer").attr("readonly", true);
    $("#chkIsimportantCustomer").attr("disabled", true);
    $("#btnViewInstalltionDetail").click(function () { ViewInstalltionDetailClick(); });

    BindExpectedOperationDate();
});

function InitialCommandControl() {
    SetRegisterCommand(true, RegisterClick);
    SetResetCommand(true, ResetClick);
}

/* --- Events --- */

function InitialEvent() {    
   
}

function BindExpectedOperationDate() {
    call_ajax_method_json('/Contract/GetExpectedOperationDate_CTS054', "",
    function (result, controls) {
        //$("#dpExpectOperationDate").datepicker("getDate");
        result.ExpectedOperationDate = ConvertDateObject(result.ExpectedOperationDate);
        $("#dpExpectOperationDate").val(ConvertDateToTextFormat(result.ExpectedOperationDate));
        if (result.InstallationStatusCode == "99") 
        {
            $("#btnViewInstalltionDetail").attr("disabled", true);
        }

    });
}

function RegisterClick() {
    VaridateCtrl(['dpExpectOperationDate'], null);
    DisableRegisterCommand(true);
    DisableResetCommand(true);
    var obj = {
        ContractCode: ObjCTS054.ContractCode,
        OCC: ObjCTS054.OCC,
        ExpectedOperationDate: $("#dpExpectOperationDate").val()
    };

    call_ajax_method_json('/Contract/ValidateALL_CTS054', obj,
    function (result, controls) {
        VaridateCtrl(['dpExpectOperationDate'], controls);
        if (result == undefined) {
            EnableConfirmCommand();
            $("#ContractChange").SetViewMode(true);
            $("#ExpectOperationDateChange").SetViewMode(true);
        }
        DisableRegisterCommand(false);
        DisableResetCommand(false);
    });

}

function ConfirmClick() {
    DisableConfirmCommand(true);
    var obj = {
                ContractCode: ObjCTS054.ContractCode,
                OCC: ObjCTS054.OCC,
                ExpectedOperationDate: $("#dpExpectOperationDate").val()
              };

              call_ajax_method_json('/Contract/ValidateALL_CTS054', obj,
              function (result, controls) {
                  if (result == undefined) {
                      call_ajax_method_json('/Contract/ConfirmClick_CTS054', obj, function (result, controls) {
                          if (result != undefined) {
                              OpenInformationMessageDialog(result.Code, result.Message, function () { window.location.href = generate_url("/common/CMS020"); });
                          }
                      });
                  }                 
              });
}

function ResetClick() {
    DisableRegisterCommand(true);
    DisableResetCommand(true);
    /* --- Get Message --- */
    var obj = {
        module: "Common",
        code: "MSG0038"
    };
    call_ajax_method_json("/Shared/GetMessage", obj, function (result) {
        DisableRegisterCommand(false);
        DisableResetCommand(false);
        OpenOkCancelDialog(result.Code, result.Message,
        function () {
            call_ajax_method_json('/Contract/ResetClick_CTS054', ObjCTS054,
              function (result, controls) {
                  if (result != null) {
                      //$("#dpExpectOperationDate").val(result);
                      EnableRegisterCommand();
                      ReBindingData(result);
                  }
              });
        },
        null);
    });
  }

  function ReBindingData(dataObj) {
      ObjCTS054 = { ContractCodeShort: dataObj.ContractCodeShort
        , ContractCode: dataObj.ContractCode
        , OCC: dataObj.OCC
        , ContractStatus: dataObj.ContractStatus
        , QuotationTargetCode: dataObj.QuotationTargetCode
        , Alphabet: ""
        , EmpNo: ""
        , EmpName: ""
        , DisplayAll: ""
        , BillingClientCode: ""
        , BillingOffice: ""
        , PaymentMethod: ""
        , Sequence: ""
        , ServiceTypeCode: dataObj.ServiceTypeCode
        , TargetCodeType: dataObj.TargetCodeType
        , EndContractDate: dataObj.EndContractDate
        , InstallationStatusCode: dataObj.InstallationStatusCode
      }

      $('#txtContractCode').val(dataObj.ContractCodeShort);
      $('#txtUserCode').val(dataObj.UserCode);
      $('#txtCustomerCode').val(dataObj.CustomerCode);
      $('#txtCustomerCodeReal').val(dataObj.RealCustomerCode);
      $('#txtSiteCode').val(dataObj.SiteCode);
      if (dataObj.ImportantFlag) {
          $('#chkIsimportantCustomer').attr('checked', 'checked');
      } else {
          $('#chkIsimportantCustomer').removeAttr('checked');
      }
      $('#txtContractTargetName').val(dataObj.CustFullNameEN);
      $('#txtContractTargetAddress').val(dataObj.AddressFullEN);
      $('#txtSiteName').val(dataObj.SiteName);
      $('#txtSiteAddress').val(dataObj.SiteAddress);
      $('#txtCustFullNameLC').val(dataObj.CustFullNameLC);
      $('#txtContractTargetAddressLocal').val(dataObj.AddressFullLC);
      $('#txtSiteNameLocal').val(dataObj.SiteNameLC);
      $('#txtSiteAddressLocal').val(dataObj.SiteAddressLC);
      $('#txtInstallationStatus').val(dataObj.InstallationStatus);
      $('#txtOperationOffice').val(dataObj.OfficeName);
      $('#QuotationTargetCode').val(dataObj.QuotationTargetCode);
      $("#dpExpectOperationDate").val(dataObj.ExpectOperationDate);
  }

function BackClick() {
    DisableConfirmCommand(true);
    /* --- Set Command Button --- */
    EnableRegisterCommand();
    $("#ContractChange").SetViewMode(false);
    $("#chkIsimportantCustomer").attr("disabled", true);
    $("#chkIsimportantCustomer").attr("readonly", true);

    $("#ExpectOperationDateChange").SetViewMode(false);

    DisableConfirmCommand(false);
}

function ViewInstalltionDetailClick() {
    //$("#dlgCTS055").OpenCMS180Dialog('CTS054');
    var obj = {
        ContractCode: $('#txtContractCode').val()
    };
    ajax_method.CallScreenControllerWithAuthority("/Common/CMS180", obj, true);
}

function EnableRegisterCommand() {
    DisableAllCommand();
    SetRegisterCommand(true, RegisterClick);
    SetResetCommand(true, ResetClick);
}

function EnableConfirmCommand() {
    DisableAllCommand();
    SetConfirmCommand(true, ConfirmClick);
    SetBackCommand(true, BackClick);
}

function DisableAllCommand() {
    SetRegisterCommand(false, null);
    SetResetCommand(false, null);
    SetConfirmCommand(false, null);
    SetBackCommand(false, null);
}

function CMS180Object() {
    return {
        strCallerScreenID: "CTS055",
        ContractCode: $("#lblContractCode").val()       
    };
}

File: SECOM_AJIS.Presentation.Contract/Models/CTS054.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations;

using SECOM_AJIS.Common.Util;
using SECOM_AJIS.Common.CustomAttribute;
using SECOM_AJIS.Presentation.Contract.Models.MetaData;
using SECOM_AJIS.Presentation.Contract.Models;
using SECOM_AJIS.Common.Models;
using SECOM_AJIS.DataEntity.Contract;

namespace SECOM_AJIS.Presentation.Contract.Models
{
    /// <summary>
    /// Parameter of CTS054 screen
    /// </summary>
    public class CTS054_Parameter 
    {
        public string ContractCode { get; set; }        
    }

    /// <summary>
    /// DO for validate business
    /// </summary>
    [MetadataType(typeof(CTS054_DOValidateBusiness_MetaData))]
    public class CTS054_DOValidateBusiness
    {
        public DateTime? ExpectedOperationDate { get; set; }
        public String ContractCode { get; set; }
        public String OCC { get; set; }
        public String InstallationStatusCode { get; set; }
    }

    /// <summary>
    /// DO of ScreenOutput
    /// </summary>
    public class CTS054_ScreenOutputObject
    {
        public string ContractCodeShort { get; set; }
        public string ContractCode { get; set; }
        public string OCC { get; set; }
        public string ContractStatus { get; set; }
        public string Alphabet { get; set; }
        public string EmpNo { get; set; }
        public string EmpName { get; set; }
        public string DisplayAll { get; set; }
        public string BillingClientCode { get; set; }
        public string BillingOffice { get; set; }
        public string PaymentMethod { get; set; }
        public string Sequence { get; set; }
        public string ServiceTypeCode { get; set; }
        public string TargetCodeType { get; set; }
        public string EndContractDate { get; set; }
        public string InstallationStatusCode { get; set; }
        public string UserCode { get; set; }
        public string CustomerCode { get; set; }
        public string RealCustomerCode { get; set; }
        public string SiteCode { get; set; }
        public string CustFullNameEN { get; set; }
        public string AddressFullEN { get; set; }
        public string SiteName { get; set; }
        public string SiteAddress { get; set; }
        public string CustFullNameLC { get; set; }
        public string AddressFullLC { get; set; }
        public string SiteNameLC { get; set; }
        public string SiteAddressLC { get; set; }
        public string InstallationStatus { get; set; }
        public string OfficeName { get; set; }
        public string QuotationTargetCode { get; set; }
        public string ExpectOperationDate { get; set; }
        public string NegotiationStaffEmpNo1 { get; set; }
        public bool ImportantFlag { get; set; }

    }

    /// <summary>
    /// Parameter of CTS054 screen
    /// </summary>
    public class CTS054_ScreenParameter : ScreenSearchParameter
    {
        public dsRentalContractData DSRentalContract { get; set; }

        [KeepSession]
        public CTS054_Parameter ScreenParameter { get; set; }

        public CTS054_DOValidateBusiness DORegisterData { get; set; }
        public CTS054_DOValidateBusiness DOValidateBusiness { get; set; }

        [KeepSession]
        public string ContractCode { get; set; }
    }
}

namespace SECOM_AJIS.Presentation.Contract.Models.MetaData
{
    public class CTS054_DOValidateBusiness_MetaData : CTS054_DOValidateBusiness
    {
        [NotNullOrEmpty]
        public string ContractCode { get; set; }

        [NotNullOrEmpty()]
        public string OCC { get; set; }

        //[NotNullOrEmpty(ControlName = "dpExpectOperationDate", Module = MessageUtil.MODULE_COMMON, MessageCode = MessageUtil.MessageList.MSG0007, Parameter = "Expect Operation Date")]
        public string ExpectedOperationDate { get; set; }
    }
}